### PR TITLE
Fix double delete on use action form

### PR DESCRIPTION
**Pull request by infiton**
we had a nasty issue with how we determined which ids have been updated/created/deleted when there are grandchildren. The ids for each array element were being addressed by the array index which the form can change (by reordering the children or deleting and appending)

This PR fixes that by addressing the existing ids not by array index but by the id of the array element

## PR Checklist

- [ ] Important or complicated code is tested
- [ ] Any user facing changes are documented in the Gadget-side changelog
- [ ] Any immediate changes are slated for release in Gadget via a generated package dependency bump
- [ ] Versions within this monorepo are matching and there's a valid upgrade path

### CODE REVIEW

File: packages/react/src/use-action-form/utils.ts [hunk 1]
```diff
@@ -365,7 +388,7 @@ export const reshapeDataForGraphqlApi = async (client: AnyClient, defaultValues:
 
       const inputHasId = "id" in input;
       const inputHasMoreFields = Object.keys(input).filter((field) => field !== "__typename").length > 1;
-      const inputUpdateId = updates[path!]?.[0];
+      const inputUpdateId = idPath && recordIdsAtPath[idPath] ? recordIdsAtPath[idPath][0] : undefined;
```
Comment (MillanWangGadget) [inline packages/react/src/use-action-form/utils.ts:391]: Consider adding a check for `recordIdsAtPath[idPath].length` in the off chance that `recordIdsAtPath[idPath]` is an empty array

File: packages/react/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gadgetinc/react",
-  "version": "0.18.4",
+  "version": "0.18.5",
```
[the remark this answers is not in the document]
Comment (infiton) [reply packages/react/package.json:3]: I'll remove the version bump and then we can merge that PR on top

File: packages/react/src/use-action-form/utils.ts (hunk 1 above)
Comment (infiton) [reply packages/react/src/use-action-form/utils.ts:391]: :+1:

**Comment by infiton** [comment]
going to merge; but this won't be released until https://github.com/gadget-inc/js-clients/pull/704 merges

**Pull request merged**